### DISCUSSION TITLE
Data template is not working as expected

### DISCUSSION THREAD

**Discussion by JyotiPMallick** [pactumjs/pactum#353] [Q&A] [answered]
I have my test framework which have few suites, for example `componenttest`. Where I have few api's for post request type. I am trying to use the data template but getting error. 

example of the spec file:

```
 it("02: Create New User : Expected 200 Status Code", async () => {
    const response = await spec()
      .post(`${configAPIEndPoint.BASE_URL}/common/jobs`)
      .withBearerToken(bearerTOKEN)
      .withHeaders("Content-Type", "application/json")
      .withBody(
       `{'@DATA:TEMPLATE@' : 'CommonJobs',}` 
      )
      .expectStatus(200)
      .expectBodyContains("message", "Success!");

      console.log(response)
  });
```

Now my body accept a JSON file and which I have used as data template, 

```
const { stash, spec } = require("pactum");

stash.addDataTemplate({
    'CommonJobs' : {
        "title"             : "Automation Test Suite Sample User 1",
        "location"          : "Hyderabad",
        "jobDescription"    : "Validation Engineer 1",
        "jobRequirements"   : "BE",
        "experience"        : "Experienced"
    }
});

```

The only difference is the spec file is present is `../test/componenttest/usermanagement.spec.js` and the data template file is present in `../test/data/componentTest.DataSet.js`

**Comment by ASaiAnudeep** [comment, marked as answer]
Hello @JyotiPMallick ,

Couple of things,
- try using `withJson({ '@DATA:TEMPLATE@': 'CommonJobs' })` 
- try loading the file at the start of test execution - Ex:  https://github.com/pactumjs/pactum-examples/blob/main/data-management/tests/_base.spec.js

**Comment by JyotiPMallick** [comment]
Thank you for the answer @ASaiAnudeep ,  just a follow up question if my request parameter is strict on `withBody` will it work ?

**Reply by ASaiAnudeep** [reply]
It should work

**Comment by JyotiPMallick** [comment]
It worked like a charm, closing this conversation
